internal/kubernetes/config.go

```diff
@@ -59,6 +74,48 @@ func (c Config) Build(shardingConfig sharding.Config) ([]promcfg.Job, error) {
 	return promScrapeJobs, nil
 }
 
+func buildTargetFilters(filters CuratedExperience, jobPrefix string, metadataSourcePrefix string) []promcfg.RelabelConfig {
+	if !filters.Enabled || !containPrefix(filters.JobsPrefix, jobPrefix) {
+		return nil
+	}
+
+	sourceLabels := make([]string, 0, len(filters.SourceLabels))
+	for _, sL := range filters.SourceLabels {
+		sanitizedLabel := invalidPrometheusLabelCharRegex.ReplaceAllString(sL, "_")
+		sourceLabels = append(sourceLabels, fmt.Sprintf("%s%s_%s", metadataSourcePrefix, labelMetadata, sanitizedLabel))
+	}
+
+	if len(filters.SourceLabels) == 0 {
+		return nil
```

Comment:
should we return an error instead?

Reply:
perhaps is a good idea, fail fast, fail hard